core-processor/src/main/java/io/micronaut/inject/writer/BeanDefinitionWriter.java

```diff
@@ -312,6 +324,11 @@ public class BeanDefinitionWriter extends AbstractClassFileWriter implements Bea
             int.class,
             String.class);
 
+    private static final Method GET_EVALUATED_EXPRESSION_VALUE_FOR_CONSTRUCTOR_ARGUMENT = ReflectionUtils.getRequiredInternalMethod(
+            AbstractInitializableBeanDefinition.class,
+            "getEvaluatedExpressionValueForConstructorArgument",
+            int.class);
```

Comment:
what about fields?

Reply:
Fields are supported as well, they just don't need a separate method for it. There is a set of tests which covers injection of expressions in `@Value` into fields, setter methods and constructor